react-stopwatch-component/my-app/src/StopWatch.tsx

```diff
@@ -0,0 +1,56 @@
+import { FaPause, FaPlay } from 'react-icons/fa';
+import { useState, useRef } from 'react';
+import './StopWatch.css';
+
+export function StopWatch() {
+  const [toggle, setToggle] = useState(true);
+  const [count, setCount] = useState(0);
+
+  const idRef = useRef<NodeJS.Timeout | null>(null);
```

Comment:
Not a bad solution, but I would just `useState` instead. Usually Refs are only used if you are trying to access a DOM element. Though they can also be used like state. The difference is that with Refs you don't trigger a re-render when you set it. In this case, that's OK. So, the more I think about it, the more I think this is a reasonable solution.